### ISSUE TITLE
Incorrect output

### ISSUE THREAD

**Issue by Richardk2n** [python-lsp/python-lsp-black#18]
I am using spyder and have told it to autoformat my code with black on save (which I am assuming uses this plugin). It produces some very odd output for one of my files. I provided the full file for context. The relevant part is the very end (return statements of the last funciton). Running black via command line does not change the files. Any idea, what could cause this?
If there is any additional information, that i should provide, please ask.

The file:
```
# -*- coding: utf-8 -*-
"""
File to house decorators.

Created on Sat Oct 31 19:30:30 2020

@author: Richard Kellnberger
"""

import functools
from threading import Lock
from typing import Callable
from time import time


def locked(lock: Lock) -> Callable:
    """
    Provide a decorator for a function to be wrapped with a lock to make it thread-save.

    Parameters
    ----------
    lock : Lock
        The lock to use.

    Returns
    -------
    Callable
        The decorator function.

    """

    def decorator(function: Callable) -> Callable:
        @functools.wraps(function)
        def wrapper_decorator(*args, **kwargs):
            lock.acquire()
            returnValue = function(*args, **kwargs)
            lock.release()
            return returnValue

        return wrapper_decorator

    return decorator


def retry(timeout: float) -> Callable:
    """
    Provide a decorator for a function to be wrapped with a retry to prevent connection losses.

    Parameters
    ----------
    timeout : float
        The timeout in seconds after which the connection is dropped.

    Returns
    -------
    Callable
        The decorator function.

    """

    def decorator(function: Callable) -> Callable:
        @functools.wraps(function)
        def wrapper_decorator(*args, **kwargs):
            t0 = time()
            ex = None  # Make sure we can delete ex (==put it in namespace)
            while time() - t0 < timeout:
                try:
                    returnValue = function(*args, **kwargs)
                except Exception as e:
                    ex = e  # Removes old ex from scope (probably) and gc gets it (probably)
                    print("Lost connection, retrying.")
                else:
                    del ex  # It worked remove stored excetions
                    return returnValue
            print("lost connection completely")
            raise ex  # It finally didn't work raise exception

        return wrapper_decorator

    return decorator

```

The formatted file:
```
# -*- coding: utf-8 -*-
"""
File to house decorators.

Created on Sat Oct 31 19:30:30 2020

@author: Richard Kellnberger
"""

import functools
from threading import Lock
from typing import Callable
from time import time


def locked(lock: Lock) -> Callable:
    """
    Provide a decorator for a function to be wrapped with a lock to make it thread-save.

    Parameters
    ----------
    lock : Lock
        The lock to use.

    Returns
    -------
    Callable
        The decorator function.

    """

    def decorator(function: Callable) -> Callable:
        @functools.wraps(function)
        def wrapper_decorator(*args, **kwargs):
            lock.acquire()
            returnValue = function(*args, **kwargs)
            lock.release()
            return returnValue

        return wrapper_decorator

    return decorator


def retry(timeout: float) -> Callable:
    """
    Provide a decorator for a function to be wrapped with a retry to prevent connection losses.

    Parameters
    ----------
    timeout : float
        The timeout in seconds after which the connection is dropped.

    Returns
    -------
    Callable
        The decorator function.

    """

    def decorator(function: Callable) -> Callable:
        @functools.wraps(function)
        def wrapper_decorator(*args, **kwargs):
            t0 = time()
            ex = None  # Make sure we can delete ex (==put it in namespace)
            while time() - t0 < timeout:
                try:
                    returnValue = function(*args, **kwargs)
                except Exception as e:
                    ex = e  # Removes old ex from scope (probably) and gc gets it (probably)
                    print("Lost connection, retrying.")
                else:
                    del ex  # It worked remove stored excetions
                    return returnValue
            print("lost connection completely")
            raise ex  # It finally didn't work raise exception

        return wrapper_decorator

    return decorator
ork raise exception

        return wrapper_decorator

    return decorator

```

Installed packages:
C:\Windows\system32>pip list
Package                           Version
--------------------------------- ---------
alabaster                         0.7.12
appdirs                           1.4.4
arrow                             1.1.1
astroid                           2.6.2
async-generator                   1.10
atomicwrites                      1.4.0
attrs                             21.2.0
autopep8                          1.5.5
Babel                             2.9.1
backcall                          0.2.0
backports.entry-points-selectable 1.1.0
bcrypt                            3.2.0
binaryornot                       0.4.4
black                             21.7b0
bleach                            3.3.1
certifi                           2021.5.30
cffi                              1.14.6
cfgv                              3.3.0
chardet                           4.0.0
charset-normalizer                2.0.3
click                             8.0.1
cloudpickle                       1.6.0
colorama                          0.4.4
cookiecutter                      1.7.3
cryptography                      3.4.7
cycler                            0.10.0
debugpy                           1.3.0
decorator                         5.0.9
defusedxml                        0.7.1
diff-match-patch                  20200713
distlib                           0.3.2
docutils                          0.17.1
entrypoints                       0.3
esc-ls                            0.1.6
filelock                          3.0.12
flake8                            3.9.2
h5py                              3.3.0
identify                          2.2.11
idna                              3.2
imagesize                         1.2.0
importlib-metadata                4.6.1
inflection                        0.5.1
intervaltree                      3.1.0
ipykernel                         6.0.2
ipython                           7.25.0
ipython-genutils                  0.2.0
isort                             5.9.2
jedi                              0.17.2
Jinja2                            3.0.1
jinja2-time                       0.2.0
jsonschema                        3.2.0
jupyter-client                    6.1.12
jupyter-core                      4.7.1
jupyterlab-pygments               0.1.2
keyring                           23.0.1
kiwisolver                        1.3.1
lazy-object-proxy                 1.6.0
MarkupSafe                        2.0.1
matplotlib                        3.4.2
matplotlib-inline                 0.1.2
mccabe                            0.6.1
mistune                           0.8.4
mypy                              0.910
mypy-extensions                   0.4.3
nbclient                          0.5.3
nbconvert                         6.1.0
nbformat                          5.1.3
nest-asyncio                      1.5.1
nodeenv                           1.6.0
numpy                             1.21.0
numpydoc                          1.1.0
packaging                         21.0
pandocfilters                     1.4.3
paramiko                          2.7.2
parso                             0.7.0
pathspec                          0.8.1
pexpect                           4.8.0
pickleshare                       0.7.5
Pillow                            8.3.1
pip                               21.2.4
platformdirs                      2.0.2
pluggy                            0.13.1
poyo                              0.5.0
pre-commit                        2.13.0
prompt-toolkit                    3.0.19
psutil                            5.8.0
ptyprocess                        0.7.0
pycodestyle                       2.7.0
pycparser                         2.20
pydocstyle                        6.1.1
pyflakes                          2.3.1
Pygments                          2.9.0
pylint                            2.9.3
pyls-spyder                       0.4.0
pylsp-mypy                        0.5.2
PyNaCl                            1.4.0
pyparsing                         2.4.7
PyQt5                             5.12.3
PyQt5-sip                         12.9.0
PyQtWebEngine                     5.12.1
pyrsistent                        0.18.0
python-dateutil                   2.8.2
python-jsonrpc-server             0.4.0
python-lsp-black                  1.0.0
python-lsp-jsonrpc                1.0.0
python-lsp-server                 1.2.1
python-slugify                    5.0.2
pytz                              2021.1
pywin32                           301
pywin32-ctypes                    0.2.0
PyYAML                            5.4.1
pyzmq                             22.1.0
QDarkStyle                        3.0.2
qstylizer                         0.2.0
QtAwesome                         1.0.3
qtconsole                         5.1.1
QtPy                              1.9.0
regex                             2021.7.6
requests                          2.26.0
rope                              0.19.0
Rtree                             0.9.7
scipy                             1.7.0
setuptools                        56.0.0
six                               1.16.0
snowballstemmer                   2.1.0
sortedcontainers                  2.4.0
Sphinx                            4.1.1
sphinxcontrib-applehelp           1.0.2
sphinxcontrib-devhelp             1.0.2
sphinxcontrib-htmlhelp            2.0.0
sphinxcontrib-jsmath              1.0.1
sphinxcontrib-qthelp              1.0.3
sphinxcontrib-serializinghtml     1.1.5
spyder                            5.1.1
spyder-kernels                    2.1.0
testpath                          0.5.0
text-unidecode                    1.3
textdistance                      4.2.1
three-merge                       0.1.1
tinycss2                          1.1.0
toml                              0.10.2
tomli                             1.0.4
tornado                           6.1
traitlets                         5.0.5
types-requests                    2.25.0
types-setuptools                  57.0.0
typing-extensions                 3.10.0.0
ujson                             4.0.2
urllib3                           1.26.6
virtualenv                        20.6.0
watchdog                          2.1.3
wcwidth                           0.2.5
webencodings                      0.5.1
wheel                             0.36.2
wrapt                             1.12.1
yapf                              0.31.0
zipp                              3.5.0

**Comment by haplo** [comment]
@Richardk2n I'm afraid there are too many unknowns to know what is happening here for sure, but I will try to give some ideas:

Are you using a virtualenv, or somehow could be running different instances of _black_? Check that the black used by the LSP plugin and the one you run manually are in the same version (`black --version`)?

I'm not familiar with spyder, how do you configure the format-on-save exactly? _python-lsp-black_ supports formatting the whole document but also a range, but black doesn't work well in this second case and it has limitations, I don't know if that could be your case.

Another idea is to try downgrading black to previous versions, it could be a regression bug in the new version.

**Comment by ccordoba12** [comment]
Hey @Richardk2n, I think this is a duplicate of spyder-ide/spyder#16180, which is fixed already and will be part of Spyder 5.1.2 version, to be released shortly.

So please wait until that version is out and let us know if the problem is solved for you.

**Comment by haplo** [comment]
Closing because of inactivity. Please comment again if this is still an issue.
